### PR TITLE
fix(sgx): remove aesm socket from the probe

### PR DESCRIPTION
**Pull request by jarkkojs**
Closes: #1618
Signed-off-by: Jarkko Sakkinen <jarkko@profian.com>

### CODE REVIEW

**Review by haraldh — CHANGES_REQUESTED** [review]
No.. I refuse.

**Comment by haraldh** [comment]
As a compromise you can remove the AESM socket check with a feature.. maybe `no_aesm_daemon`

File: src/backend/sgx/mod.rs [hunk 1]
```diff
@@ -40,7 +40,8 @@ impl crate::backend::Backend for Backend {
 
     #[inline]
     fn have(&self) -> bool {
-        self.data().iter().all(|x| x.pass)
+        //self.data().iter().all(|x| x.pass)
+        true
```
Comment (haraldh) [inline src/backend/sgx/mod.rs:44]: At least, you could have reverted to the last state:
```suggestion
        data::dev_sgx_enclave().pass
```

**Review by haraldh — CHANGES_REQUESTED** [review]

File: src/backend/sgx/mod.rs (hunk 1 above)
Comment (jarkkojs) [reply src/backend/sgx/mod.rs:44]: I think that opt-out flag is fine. I'll rework this.

**Comment by haraldh** [comment]
or just:
```
$ sudo touch /var/run/aesmd/aesm.socket
```

**Comment by jarkkojs** [comment]
I think a opt-out flag is the most convenient in the end and it does not disturb anyone else.

**Comment by jarkkojs** [comment]
```
    Finished test [unoptimized + debuginfo] target(s) in 0.23s
     Running unittests src/main.rs (target/debug/deps/enarx-707d01f9882c3b3e)

running 8 tests
test backend::sgx::attestation::tests::request_target_info ... ignored
test backend::sev::snp::tests::test_const_id_macro ... ok
test backend::sgx::ioctls::tests::restrict_permissions ... ok
test backend::sev::snp::firmware::test::test_vcek_url ... ok
test backend::sgx::ioctls::tests::relax_permissions ... ok
test cli::snp::tests::test_empty_cache_path ... ok
test workldr::wasmldr::test::is_builtin ... ok
test cli::snp::tests::test_get_or_write ... ok

test result: ok. 7 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out; finished in 0.20s

     Running tests/c_integration_tests.rs (target/debug/deps/c_integration_tests-6a1b1a64ae866b1a)

running 21 tests
test get_att ... ignored
test bind ... ok
test clock_gettime ... ok
test exit_one ... ok
test close ... ok
test exit_zero ... ok
test getegid ... ok
test sgx_get_att_quote ... ignored
test sgx_get_att_quote_size ... ignored
test geteuid ... ok
test getgid ... ok
test write_emsgsize ... ignored
test write_stderr ... ignored
test getuid ... ok
test listen ... ok
test read ... ok
test read_udp ... ok
test readv ... ok
test socket ... ok
test uname ... ok
test write_stdout ... ok

test result: ok. 16 passed; 0 failed; 5 ignored; 0 measured; 0 filtered out; finished in 45.16s

     Running tests/rust_integration_tests.rs (target/debug/deps/rust_integration_tests-d6004f79e1daf288)

running 6 tests
test rust_sev_attestation ... ignored
test unix_echo ... ignored
test echo ... ok
test memory_stress_test ... ok
test cpuid ... ok
test memspike ... ok

test result: ok. 4 passed; 0 failed; 2 ignored; 0 measured; 0 filtered out; finished in 13.24s

     Running tests/wasmldr_tests.rs (target/debug/deps/wasmldr_tests-bb7dca85755c838b)

running 9 tests
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test check_tcp ... FAILED
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test memory_stress_test ... FAILED
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test hello_wasi_snapshot1 ... FAILED
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test memspike ... FAILED
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test no_export ... ok
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test return_1 ... FAILED
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test wasi_snapshot1 ... FAILED
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test zerooneone ... FAILED
Error: error obtaining attestation key id

Caused by:
    No such file or directory (os error 2)
test echo ... FAILED

failures:

---- check_tcp stdout ----
thread 'check_tcp' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: Expected exit status differs.', tests/common/mod.rs:191:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- memory_stress_test stdout ----
thread 'memory_stress_test' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: Expected exit status differs.', tests/common/mod.rs:191:5

---- hello_wasi_snapshot1 stdout ----
thread 'hello_wasi_snapshot1' panicked at 'assertion failed: `(left == right)`
  left: `[]`,
 right: `[72, 101, 108, 108, 111, 44, 32, 119, 111, 114, 108, 100, 33, 10]`: Expected contents of stdout output differs', tests/common/mod.rs:170:13

---- memspike stdout ----
thread 'memspike' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: Expected exit status differs.', tests/common/mod.rs:191:5

---- return_1 stdout ----
thread 'return_1' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: Expected exit status differs.', tests/common/mod.rs:191:5

---- wasi_snapshot1 stdout ----
thread 'wasi_snapshot1' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: Expected exit status differs.', tests/common/mod.rs:191:5

---- zerooneone stdout ----
thread 'zerooneone' panicked at 'assertion failed: `(left == right)`
  left: `[]`,
 right: `[84, 98, 98, 113, 32, 122, 98, 101, 97, 118, 97, 116, 44, 32, 103, 117, 110, 103, 39, 102, 32, 110, 32, 97, 118, 112, 114, 32, 103, 97, 114, 103, 114, 97, 97, 111, 110, 46, 10, 48, 49, 49, 56, 32, 57, 57, 57, 32, 56, 56, 49, 32, 57, 57, 57, 32, 49, 49, 57, 32, 55, 50, 53, 32, 51, 10]`: Expected contents of stdout output differs', tests/common/mod.rs:170:13

---- echo stdout ----
thread '<unnamed>' panicked at 'failed to write stdin to child: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }', tests/common/mod.rs:122:18
thread 'echo' panicked at 'failed to provide input for process `echo`', tests/common/mod.rs:139:13


failures:
    check_tcp
    echo
    hello_wasi_snapshot1
    memory_stress_test
    memspike
    return_1
    wasi_snapshot1
    zerooneone

test result: FAILED. 1 passed; 8 failed; 0 ignored; 0 measured; 0 filtered out; finished in 33.13s

error: test failed, to rerun pass '--test wasmldr_tests'
```

This is what I get now.

**Comment by jarkkojs** [comment]
That does not matter to me tho, clearly the tests actually pass, and attestation fails.

**Review by haraldh — APPROVED** [review]

**Comment by jarkkojs** [comment]
I updated with a fix for `wasmldr`, which does not work yet because I don't how the feature flag goes through.

**Comment by npmccallum** [comment]
@jarkkojs I'm rejecting this PR. We don't want additional features. If you're running on SGX, AESMD is a prereq.

**Pull request closed without merging**